contracts/CozyInvestConvex.sol

```diff
@@ -0,0 +1,223 @@
+
+  function remove_liquidity_one_coin(
+    uint256 amount,
+    int128 index,
+    uint256 minAmount
+  ) external returns (uint256);
+
+  function remove_liquidity_one_coin(
+    address pool,
+    uint256 amount,
+    int128 index,
+    uint256 minAmount
+  ) external returns (uint256);
+}
+
+/**
+ * @notice On-chain scripts for borrowing from Cozy and using the borrowed funds to
+ * supply to curve and then deposit into Convex
+ * @dev This contract is intended to be used by delegatecalling to it from a DSProxy
+ */
+contract CozyInvestConvex is CozyInvestHelpers, ICozyInvest1, ICozyDivest1, ICozyReward {
+  using SafeERC20 for IERC20;
+
+  /// @notice The unprotected money market we borrow from / repay to
+  address public immutable moneyMarket;
+  /// @notice The protection market we borrow from / repay to
+  address public immutable protectionMarket;
+  /// @notice Underlying of the above money market and protection market
+  address public immutable underlying;
+  /// @notice Curve deposit zap
+  address public immutable curveDepositZap;
+  /// @notice Token index for the curve pool
+  int128 public immutable curveIndex;
+  /// @notice Curve LP token
+  address public immutable curveLpToken;
+  /// @notice Convex booster id
+  uint256 public immutable convexPoolId;
+  /// @notice Convex reward manager
+  address public immutable convexRewardManager;
+
+  /// @notice Curve metapool
+  bool public immutable longSigFormat;
+
+  /// @notice Convex deposit contract address
+  address public constant convex = 0xF403C135812408BFbE8713b5A23a04b3D48AAE31;
+
+  /// @notice Curve DAI/USDC/USDT (3Crv) pool
+  address public constant tricurve = 0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490;
+
+  constructor(
+    address _moneyMarket,
+    address _protectionMarket,
+    address _underlying,
+    address _curveDepositZap,
+    uint256 _convexPoolId,
+    int128 _curveIndex,
+    bool _longSigFormat
+  ) {
+    moneyMarket = _moneyMarket;
+    protectionMarket = _protectionMarket;
+    underlying = _underlying;
+    curveDepositZap = _curveDepositZap;
+    convexPoolId = _convexPoolId;
+    longSigFormat = _longSigFormat;
+    curveIndex = _curveIndex;
+
+    (address _curveLpToken, , , address _convexRewardManager, , ) = IConvexBooster(convex).poolInfo(convexPoolId);
+    curveLpToken = _curveLpToken;
+    convexRewardManager = _convexRewardManager;
+  }
+
+  /**
+   * @notice Invest method for borrowing underlying and depositing it to Curve and then Convex
+   * @param _market Address of the market to repay
+   * @param _borrowAmount Amount of underlying to borrow and invest
+   * @param _curveMinAmountOut The minimum amount we expect to receive when adding liquidity to Curve
+   */
+  function invest(
+    address _market,
+    uint256 _borrowAmount,
+    uint256 _curveMinAmountOut
+  ) external {
+    // 1. Borrow underlying from cozy
+    require((_market == address(moneyMarket) || _market == address(protectionMarket)), "Invalid borrow market");
```

Comment:
extra parentheses here and in `divest` also (I see that's also true in the original this is copied from, but may as well fix it here)